### ISSUE TITLE
A question about parallel_import 

### ISSUE THREAD

**Issue by myeong** [uwescience/myria#837]
I ran the following code: 

```
connection = MyriaConnection(rest_url='http://ec2-54-200-153-177.us-west-2.compute.amazonaws.com:8753', execution_url='http://ec2-54-200-153-177.us-west-2.compute.amazonaws.com:8080')
schema = MyriaSchema({"columnTypes" : ["LONG_TYPE", "LONG_TYPE", "LONG_TYPE", "FLOAT_TYPE"], "columnNames" : ["time", "source", "dest", "call"]})
relation = MyriaRelation('parallelLoad', connection=connection, schema=schema)

# A list of worker-URL pairs -- must be one for each worker
work = [(1, 'https://s3-us-west-2.amazonaws.com/census-cdr/test'),
       (2, 'https://s3-us-west-2.amazonaws.com/census-cdr/test'),
       (3, 'https://s3-us-west-2.amazonaws.com/census-cdr/test'),
       (4, 'https://s3-us-west-2.amazonaws.com/census-cdr/test')]

# Upload the data
query = MyriaQuery.parallel_import(relation=relation, work=work)
print query.status
```

and the output is 
`RUNNING`

I don't understand how the parallel_import work, and what's the difference between this and using 'load' query from MyriaL.

1. what does "worker-URL pairs" means? Do I have to put the same dataset for all the workers, or manually put different datasets to each worker?
2. If I just load the data using MyriaL's "load" query, it doesn't load the data in parallel across the workers?
3. How do I know when the loading finishes? for now I don't see any changes in the Myria's datasets interface (http://ec2-54-200-153-177.us-west-2.compute.amazonaws.com:8080/datasets)  
4. What's the name of the table after this operation? 

After the experiment, I have to load 300GB of data onto multiple workers, so I want to know how "work" variable should be set, and whether the data are correctly loading from S3. 

Thanks!

**Comment by senderista** [comment]
@BrandonHaynes could you explain how `parallel_import` works?

**Comment by BrandonHaynes** [comment]
Hi Myeong --

These are great questions!

1. The `parallel_load` function accepts a set of (`node`-`filename`) pairs.  So if you have four workers and want to load a dataset, you should split it into four files and specify `(n, filename-n)` in the invocation.  You can use the function `query.wait_for_completion()` to block until the operation completes. 

2. Currently the MyriaL `load` command will load a dataset using a single worker, and so may not scale well for very large datasets.  We're in the final stages of extending smart parallel load support to this function, so stay tuned!

3. While you can continue to monitor `query.status` using Python, the easiest way to see the status of a running query is from the "Query" tab of the website.  See the [http://demo.myria.cs.washington.edu/queries](query tab) on the demo site for an example.  You can click on the number next to each query to view it's status and/or result.

    For example, when I looked at the status for your parallel ingest query earlier today, I saw in the exception details that one of the files you specified was not found.

4. The `parallel_ingest` function accepts a `relation` argument, which specifies where the data is to be stored.  In your example above you used the name  `parallelLoad`, and so the data will be stored in `public:adhoc:parallelLoad`.  You'll be able to see this on the website, but only if the query is successful!

Hope this helps!  For quicker response, please feel free to reach out to us on our email alias myria-users@cs.washington.edu!

Brandon
